lib/reserved-words.js

```diff
@@ -0,0 +1,177 @@
+var assert = require('assert');
+
+/**
+ * Structure for storing keywords.
+ *
+ * @typedef {Object.<String,Boolean>} KeywordsHash
+ */
+
+/**
+ * ECMAScript dialects.
+ *
+ * @const
+ * @type {Object.<String,Number|String>} - keys as readable names and values as versions
+ */
+var DIALECTS = {
+    es3: 3,
+    es5: 5,
+    es2015: 6,
+    es7: 7,
+
+    // aliases
+    es6: 6,
+    'default': 5,
+    next: 6
+};
+
+/**
+ * ECMAScript reserved words.
+ *
+ * @type {Object.<String,KeywordsHash>}
+ */
+var KEYWORDS = exports.KEYWORDS = {};
+
+/**
+ * Check word for being an reserved word.
+ *
+ * @param {String} word - word to check
+ * @param {String|Number} [dialect] - dialect or version
+ * @param {Boolean} [strict] - strict mode
+ * @returns {?Boolean}
+ */
+exports.check = function check(word, dialect, strict) {
```

Comment:
This looks pretty great, i would even move this to standalone module, since it might be useful for other projects as well.
It seems though `esutils` module already does this job, although with different interface.